specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json

```diff
@@ -0,0 +1,1120 @@
+        },
+        "CreatedTime": {
+          "type": "string",
+          "description": "The created time"
+        },
+        "CreatedBy": {
+          "type": "string",
+          "description": "The created by"
+        },
+        "ActionDetails": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ActionDetail"
+          },
+          "description": "The list of action detail"
+        }
+      },
+      "required": [
+        "State"
+      ]
+    },
+    "ActionDetail": {
+      "description": "The action detail",
+      "type": "object",
+      "properties": {
+        "MechanismType": {
+          "type": "string",
+          "description": "The mechanism type"
+        },
+        "Name": {
+          "type": "string",
+          "description": "The name of the action"
+        },
+        "Status": {
+          "type": "string",
+          "description": "The status of the action"
+        },
+        "SubState": {
+          "type": "string",
+          "description": "The substatus of the action"
+        },
+        "SendTime": {
+          "type": "string",
+          "description": "The send time"
+        },
+        "Detail": {
+          "type": "string",
+          "description": "The detail"
+        }
+      }
+    },
+    "Context": {
+      "description": "The context info",
+      "type": "object",
+      "properties": {
+        "NotificationSource": {
+          "type": "string",
+          "description": "The source of the notification request"
+        },
+        "ContextType": {
+          "type": "string",
+          "description": "The context id type"
+        }
+      }
+    },
+    "ActionGroupResource": {
+      "description": "An action group resource.",
+      "type":"object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ActionGroup",
+          "description": "The action groups properties of the resource."
+        }
+      }
+    },
+    "ActionGroupList": {
+      "description": "A list of action groups.",
+      "type":"object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ActionGroupResource"
+          },
+          "description": "The list of action groups."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Provides the link to retrieve the next set of elements."
+        }
+      }
+    },
+    "ActionGroup": {
+      "description": "An Azure action group.",
+      "type":"object",
+      "properties": {
+        "groupShortName": {
+          "type": "string",
+          "maxLength": 12,
+          "description": "The short name of the action group. This will be used in SMS messages."
+        },
+        "enabled": {
```

Comment:
>enabled [](http://example.com/codeflow?start=9&length=7)

Replace boolean properties with better enum
Check 'Replace boolean/switch properties with better enum' in [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html)

Reply:
Reading the docs, it seems like a better enum is preferred for bools in general to make it future-proof and allow multiple values if needed, but in this case, the field just indicates if the action is enabled or not, and will always have a True/False value. Is a string enum really needed in this case?